File: solutions/cpp/0209.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int ans = INT_MAX;
        int sum = 0;
        int i = 0;

        for (int j = 0; j < nums.size(); j++) {
            sum += nums[j];
            while (sum >= s) {
                ans = min(ans, j - i + 1);
                sum -= nums[i++];
            }
        }

        return ans == INT_MAX ? 0 : ans;
    }
};

File: solutions/python3/0209.py
class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        ans = float('inf')
        sum = 0
        i = 0

        for j in range(len(nums)):
            sum += nums[j]
            while sum >= s:
                ans = min(ans, j - i + 1)
                sum -= nums[i]
                i += 1

        return 0 if ans == float('inf') else ans
